File: src/features/estimates/dto/create-estimate.dto.ts
import { Type } from 'class-transformer';
import { IsArray, IsOptional, IsString, ValidateNested } from 'class-validator';
import { EstimateLineItemDto } from './estimate-line-item.dto';

export class CreateEstimateDto {
  @IsString()
  name: string;

  @IsOptional()
  @IsString()
  description?: string;

  @IsOptional()
  @IsString()
  notes?: string;

  @IsOptional()
  @IsArray()
  @ValidateNested({ each: true })
  @Type(() => EstimateLineItemDto)
  lineItems?: EstimateLineItemDto[];
}

File: spring-boot-api/src/main/java/com/mytegroup/api/dto/estimates/CreateEstimateDto.java
package com.mytegroup.api.dto.estimates;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;

import java.util.List;

public record CreateEstimateDto(
    @NotBlank(message = "Name is required")
    String name,
    String description,
    String notes,
    @Valid
    List<EstimateLineItemDto> lineItems
) {
    public CreateEstimateDto {
        if (name != null) {
            name = name.trim();
        }
        if (description != null) {
            description = description.trim();
        }
        if (notes != null) {
            notes = notes.trim();
        }
    }
}


